x-pack/plugin/rank-rrf/src/test/java/org/elasticsearch/xpack/rank/linear/LinearRetrieverBuilderTests.java

```diff
@@ -0,0 +1,329 @@
+            null,
+            List.of("field_*^1.5", "*_field_1^2.5"),
+            "baz",
+            MinMaxScoreNormalizer.INSTANCE,
+            DEFAULT_RANK_WINDOW_SIZE,
+            new float[0],
+            new ScoreNormalizer[0]
+        );
+        assertSimplifiedParamsRewrite(
+            retriever,
+            queryRewriteContext,
+            Map.of("field_*", 1.5f, "*_field_1", 2.5f),
+            Map.of("semantic_field_1", 2.5f),
+            "baz",
+            MinMaxScoreNormalizer.INSTANCE
+        );
+
+        // Multiple boosts defined on the same field
+        retriever = new LinearRetrieverBuilder(
+            null,
+            List.of("field_*^1.5", "field_1^3.0", "*_field_1^2.5", "semantic_*^1.5"),
+            "baz2",
+            MinMaxScoreNormalizer.INSTANCE,
+            DEFAULT_RANK_WINDOW_SIZE,
+            new float[0],
+            new ScoreNormalizer[0]
+        );
+        assertSimplifiedParamsRewrite(
+            retriever,
+            queryRewriteContext,
+            Map.of("field_*", 1.5f, "field_1", 3.0f, "*_field_1", 2.5f, "semantic_*", 1.5f),
+            Map.of("semantic_field_1", 3.75f, "semantic_field_2", 1.5f),
+            "baz2",
+            MinMaxScoreNormalizer.INSTANCE
+        );
+
+        // All-fields wildcard
+        retriever = new LinearRetrieverBuilder(
+            null,
+            List.of("*"),
+            "qux",
+            MinMaxScoreNormalizer.INSTANCE,
+            DEFAULT_RANK_WINDOW_SIZE,
+            new float[0],
+            new ScoreNormalizer[0]
+        );
+        assertSimplifiedParamsRewrite(
+            retriever,
+            queryRewriteContext,
+            Map.of("*", 1.0f),
+            Map.of("semantic_field_1", 1.0f, "semantic_field_2", 1.0f),
+            "qux",
+            MinMaxScoreNormalizer.INSTANCE
+        );
+    }
+
+    public void testSearchRemoteIndex() {
+        final ResolvedIndices resolvedIndices = createMockResolvedIndices(
+            "local-index",
+            List.of(),
+            Map.of("remote-cluster", "remote-index")
+        );
+        final QueryRewriteContext queryRewriteContext = new QueryRewriteContext(
+            parserConfig(),
+            null,
+            null,
+            resolvedIndices,
+            new PointInTimeBuilder(new BytesArray("pitid")),
+            null
+        );
+
+        LinearRetrieverBuilder retriever = new LinearRetrieverBuilder(
+            null,
+            null,
+            "foo",
+            MinMaxScoreNormalizer.INSTANCE,
+            DEFAULT_RANK_WINDOW_SIZE,
+            new float[0],
+            new ScoreNormalizer[0]
+        );
+
+        IllegalArgumentException iae = expectThrows(IllegalArgumentException.class, () -> retriever.doRewrite(queryRewriteContext));
+        assertEquals("[linear] does not support the simplified query format when querying remote indices", iae.getMessage());
+    }
+
+    private static ResolvedIndices createMockResolvedIndices(
```

Comment:
There's some duplication between this and the RRF tests, could that be cleaned up a bit?

Reply:
I'm fine with it as a followup post FF